content/articles/json-server-and-axios/index.md

```diff
@@ -0,0 +1,350 @@
+
+**Axios** is an HTTP client to make HTTP requests to the JSON server.
+
+In this article, we will build a shopping list application but start with a blank Vue.js application then add `json-server` for local data storage, and `axios` for making HTTP requests.
+
+### Table of contents
+- [New vue.js application](#new-vue.js-application)
+- [Creating a JSON file and installing json-server](#creating-a-json-file-and-installing-json-server)
+- [Getting HTTP data within our application](#getting-http-data-within-our-application)
+- [Adding data using POST](#adding-data-using-post)
+- [Updating and deleting data using PATCH and DELETE respectively](#updating-and-deleting-data-using-patch-and-delete-respectively)
+
+### Prerequisites
+- [Node.js](https://nodejs.org/en/download/package-manager/)
+- basic knowledge of javascript
+
+### New vue.js application
+Assuming you have installed [Node.js](https://nodejs.org/en/download/package-manager/) in your computer, run the following commands in your terminal to install Vue CLI:
+```bash
+#Install Vue.js CLI globally
+$ npm install -g @vue/cli
+```
+After the installation, create a new project using:
+```bash
+$ vue create shopping-list 
+```
+Navigate to the folder and open the project in your favorite editor. e.g vs code:
+```bash
+$ cd shopping-list
+
+# open project in vs code
+$ code .
+```
+
+Run the following command to start the application:
+```bash
+$ npm run serve
+```
+Our application will run on `http://localhost:8080/` in the browser. We should now have a blank Vue.js project running on our browser.
+
+### Creating a JSON file and installing `json-server`
+We can now create a simple JSON file that can be used as a database. Inside our shopping-list directory, let's create a file called data.json:
+```bash
+$ touch data.json
+```
+Let's add the following details in the file:
+```JSON
+{
+  "items":[
+      {
+      "id":0,
+      "name":"Carrots"
+      },
+      {
+      "id":1,
+      "name":"Cabbage"
+      },
+      {
+      "id":2,
+      "name":"Banana"
+      },
+      {
+      "id":0,
+      "name":"Cakes"
+      }
+  ]
+}
+```
+From the code above, we have created a shopping list but we need to install `json-server` and start it:
+```bash
+#Install json-server globally
+$ npm install json-server -g 
+
+#Inside your working directory, start the json-server
+$ json-server data.json
+```
+If we type `http://localhost:3000/items` in our browser, we should see our 'shopping list' in the database.
+
+### Getting HTTP data within our application
+First, we need to install `Axios` in our project to access the data in the `data.json`:
+```bash
+npm install axios
+```
+After the installation, we can now import axios in our application. Inside App.vue, add the following code:
+```js
+<template>
+ <div id="app">
```

Comment:
Change the language identifier to `vue` or `vuejs`. Do this to other snippets where applicaple.